plugins/serializers/msgpack/metric.go

```diff
@@ -0,0 +1,101 @@
+package msgpack
+
+import (
+	"encoding/binary"
+	"time"
+
+	"github.com/tinylib/msgp/msgp"
+)
+
+//go:generate msgp
+
+// Metric is structure to define MessagePack message format
+// will be used by msgp code generator
+type Metric struct {
+	Name   string                 `msg:"name"`
+	Time   MessagePackTime        `msg:"time,extension"`
+	Tags   map[string]string      `msg:"tags"`
+	Fields map[string]interface{} `msg:"fields"`
+}
+
+// MessagePackTime implements the official timestamp extension type
+// https://github.com/msgpack/msgpack/blob/master/spec.md#timestamp-extension-type
+//
+// tinylib/msgp has been using their own custom extension type and the official extension
+// is not available. (https://github.com/tinylib/msgp/issues/214)
+type MessagePackTime struct {
+	time time.Time
+}
+
+func init() {
+	msgp.RegisterExtension(-1, func() msgp.Extension { return new(MessagePackTime) })
+}
+
+// ExtensionType implements the Extension interface
+func (*MessagePackTime) ExtensionType() int8 {
+	return -1
+}
+
+// Len implements the Extension interface
+func (t *MessagePackTime) Len() int {
+	sec := t.time.Unix()
+	nsec := t.time.Nanosecond()
+
+	if sec < 0 || sec > 0x400000000 { // 96 bits encoding
+		return 12
+	} else if sec > 0xFFFFFFFF || nsec != 0 {
+		return 8
+	} else {
+		return 4
+	}
```

Comment:
Could you please use
```suggestion
	if sec < 0 || sec > (1 << 34) { // 96 bits encoding
		return 12
	} else if sec >= (1 << 32) || nsec != 0 {
		return 8
	} else {
		return 4
	}
```
so it's easier to understand what the bit-masks mean.

Reply:
Ack.